src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,761 @@
+                payload: queryBuilder,
+              });
+              resolve();
+            } catch (e) {
+              reject(e);
+            }
+          }, timeOutPeriod);
+        });
+      });
+
+      it("should emit a PUSH_RECORD action from the observable when called once", () => {
+        const id = random.number().toString();
+
+        const observableAction = ActionsObservable.of({
+          type: context.types.FETCH_RECORD,
+          payload: { id },
+        });
+
+        const epic = fetchRecordEpic(context);
+        const sut = epic(observableAction, store);
+
+        let action;
+        sut
+          .take(1)
+          .subscribe(a => action = a);
+
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              expect(action).to.deep.equal({
+                type: "PUSH_RECORD",
+                payload: { 
+                  fullText: expectedResult.fullText,
+                  id, 
+                },
+              });
+              resolve();
+            } catch (e) {
+              reject(e);
+            }
+          }, timeOutPeriod);
+        });
+      });
+
+    });
+
+    describe("when coalesceFindRequests is false", () => {
+      const shortTimeout = DEFAULT_COALESCE_BUFFER_TIME / 2.0;
+      const promiseTimeout = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      it("should not buffer the request", () => {
+        const id = random.number().toString();
+        const secondId = random.number().toString();
+        const sut = createBufferObservable(context);
+
+        let action;
+        sut(id)
+          .subscribe(a => action = a);
+
+        setTimeout(() => {
+          sut(secondId);
+        }, shortTimeout);
+
+        const queryBuilder = new QueryBuilder(context.name, { ids: [id, secondId] });
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              expect(action).to.not.equal({
+                type: "FETCH_ALL",
+                payload: queryBuilder,
+              });
+              resolve();
+            } catch (e) {
+              reject(e);
+            }
+          }, promiseTimeout);
+        });
+      });
+
+      it("should emit a PUSH_RECORD action for the requested item", () => {
+        const id = random.number().toString();
+        const secondId = random.number().toString();
+        
+        const sut = createBufferObservable(context);
+
+        let action;
+        sut(id)
+          .subscribe(a => action = a);
+
+        setTimeout(() => {
+          sut(secondId);
+        }, shortTimeout);
+
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              expect(action).to.not.equal({
+                type: "PUSH_RECORD",
+                payload: {
+                  fullText: expectedResult.fullText,
+                  id: secondId,
+                },
+              });
```

Comment:
I think this should be a positive test as the test's description suggests (i.e. expect the payload to have the first id) It may not be necessary or add worth to include a second request